xdsl/dialects/stablehlo.py

```diff
@@ -0,0 +1,41 @@
+from xdsl.dialects.builtin import TensorType
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, operand_def, result_def
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_op_definition
+class AbsOp(IRDLOperation):
+    name = "stablehlo.abs"
+    operand = operand_def(TensorType)
+    result = result_def(TensorType)
```

Comment:
xdsl uses IRDL Constraints to do things like verifying that some types are the same. There are a few examples of this around, in this case it would look something like this:
```suggestion
    T = Annotated[TensorType, ConstraintVar("T")]

    operand = operand_def(T)
    result = result_def(T)
```
The mismatch in either the shape or the element will be checked automatically. One thing I'm not sure about is whether the operation really requires the element type and the shape to be the same, or whether there is any broadcasting or dynamic shapes possible? A link to the doc would help disambiguate what level of abstraction this op is in.

Reply:
The [specification for this op](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#abs) states the following:

> Constraints
>
>    (C1) shape(result) = shape(operand).
>    (C2) baseline_element_type(result) is defined as:
>        complex_element_type(element_type(operand)) if is_complex(operand).
>        baseline_element_type(operand) otherwise.

I'll add a link to the documentation

Reply:
OK I think we're good with my suggestion for now, I'm not sure exactly what they mean with the complex stuff, but it sounds like we can handle it in the future

Reply:
It is just for complex<{f32,f64, ....}>

Reply:
Yeah but the actual pairs of types aren't super clear to me. Let's cross that bridge when we get to it, though, I don't think it's very likely to come up soon.

Reply:
Ah sorry, I have been just doing a lot of this without thinking, but `abs` of a complex type just computes the norm.

Reply:
Maybe I'll skip the constraint for now, since for complex values it is different.